table/column.go

```diff
@@ -286,7 +286,7 @@ func CheckOnce(cols []*Column) error {
 
 // CheckNotNull checks if nil value set to a column with NotNull flag is set.
 func (c *Column) CheckNotNull(data types.Datum) error {
-	if (mysql.HasNotNullFlag(c.Flag) && data.IsNull() || mysql.HasPreventNullInsertFlag(c.Flag)) && data.IsNull() {
```

Comment:
why `data.IsNull()` can be removed?

Reply:
Because `(a && b || c) && b = (a && b && b) || (c && b) = (a && b) || (c && b) = (a || c) && b`.

Reply:
Got it.